virtualizarr/readers/dmrpp.py

```diff
@@ -406,24 +408,33 @@ def _parse_variable(self, var_tag: ET.Element) -> Variable:
                 chunks_shape = shape
             chunkmanifest = self._parse_chunks(chunks_tag, chunks_shape)
             # Filters
-            filters = self._parse_filters(chunks_tag, dtype)
+            codecs = self._parse_filters(chunks_tag, dtype)
         # Attributes
         attrs: dict[str, Any] = {}
         for attr_tag in var_tag.iterfind("dap:Attribute", self._NS):
             attrs.update(self._parse_attribute(attr_tag))
         # Fill value is placed in zarr array's fill_value and variable encoding and removed from attributes
         encoding = {k: attrs.get(k) for k in self._ENCODING_KEYS if k in attrs}
         fill_value = attrs.pop("_FillValue", None)
-        # create ManifestArray and ZArray
-        zarray = ZArray(
-            chunks=chunks_shape,
-            dtype=dtype,
-            fill_value=fill_value,
-            filters=filters,
-            order="C",
+        # create ManifestArray
+        metadata = ArrayV3Metadata(
             shape=shape,
+            data_type=dtype,
+            chunk_grid={
+                "name": "regular",
+                "configuration": {"chunk_shape": chunks_shape},
+            },
+            chunk_key_encoding={"name": "default"},
+            fill_value=fill_value,
+            codecs=convert_to_codec_pipeline(
+                codecs=codecs,
+                dtype=dtype,
+            ),
+            attributes=attrs,
+            dimension_names=None,
```

Comment:
Do we want to specify attrs here? I feel like we want neither `dimension_names` nor `attrs`, because in virtualizarr those both are stored in the xarray objects instead.

Reply:
`dimension_names` is a required argument (even though it can be `None`, I'm not sure why `None` is not set as a default in the [__init__ method](https://github.com/zarr-developers/zarr-python/blob/v3.0.2/src/zarr/core/metadata/v3.py#L259-L260). I have removed setting attributes from this reader in https://github.com/zarr-developers/VirtualiZarr/pull/429/commits/86d1de5af91501fbdd0b0d8e2f8545645993a7d1

Should we request upstream for a default of None to be allowed in the __init__ method of ArrayV3Metadata?